modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php

```diff
@@ -86,6 +86,25 @@ public function add_iframed_body_class( $classes ) {
 		return $classes;
 	}
 
+	/**
+	 * Checks to see if cookie can be set in current context. If 3rd party cookie blocking
+	 * is enabled the editor can't load in iFrame, so emiting X-Frame-Options: DENY will
+	 * force the editor to break out of the iFrame.
+	 */
+	private function check_iframe_cookie_setting() {
+		if ( ! isset( $_SERVER['QUERY_STRING'] ) || ! strpos( $_SERVER['QUERY_STRING'], 'calypsoify%3D1%26block-editor' ) || isset( $_COOKIE['wordpress_test_cookie'] ) ) {
+			return;
+		}
+
+		if ( ! $_GET['calypsoify_cookie_check'] ) {
+			header( 'Location: ' . $_SERVER['REQUEST_URI'] . '&calypsoify_cookie_check=true' );
```

Comment:
Do you think we'd need to escape this?

Reply:
not sure about this - seems to me like it isn't needed as it is essentially just the same url that the server has already received, with no additional user generated params, etc. add, but have now added it anyway just to be safe ... what do you think?